### PR TITLE
Add support of old drafts in marketing table, add transformation of old drafts

### PR DESCRIPTION
**Pull request by radiansz**
To check this out:
1) Find an old draft here http://somethingLocal/mailings/manage (it has warning mark on icon). If you dont have one, you can create it here -- https://app.sendsay.ru/
2) Click on it, pick one of options and press the blue button.
3) After that you should find yourself on editing page of newly created entity with name 'Создано из \[name] [date]'.
In case of marketing draft some steps should not be completed. All other options creating completed entity right after pressing the blue button.

### CODE REVIEW

File: src/blocks/form/TransformOldDraft/InfoMessage.ejs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<div class="<%- this.generateClassName('InfoMessage', this.modifiers) %>">
+    <% if (this.props.type === 'marketing') { %>
+
+        <div><%=i18n('Будет создан черновик маркетингового выпуска “Создано из %draftname%”.', this.props.draft.get('name'))%></div>
```
Comment (zouxuoz) [inline src/blocks/form/TransformOldDraft/InfoMessage.ejs]: should be escaped

File: src/blocks/form/TransformOldDraft/InfoMessage.ejs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        <div><%=i18n('После создания вы будете перенаправлены на страницу редактирования созданного черновика.')%></div>
+        <div><%=i18n('Черновик “%draftname%” останется в системе и может быть удалён позднее.', this.props.draft.get('name'))%></div>
+
+    <% } else if(this.props.type === 'transactional') { %>
```
Comment (zouxuoz) [inline src/blocks/form/TransformOldDraft/InfoMessage.ejs]: missed whitespace after `if`

File: src/blocks/form/TransformOldDraft/InfoMessage.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+define(function(require) {
+    var Form = require('kit/block/block');
+    var template = require('tpl!./InfoMessage.ejs');
+    
+    return Form.extend({
+        template: template,
```
Comment (zouxuoz) [inline src/blocks/form/TransformOldDraft/InfoMessage.js]: unnecessary comma

File: src/blocks/form/TransformOldDraft/index.ejs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    var RadioButton = require('blocks/radioButton/radioButton');
+    var Button = require('blocks/button/button');
+    var InfoMessage = require('./InfoMessage');
+    var DraftPreview = require('./ScaledDraftPreview');
+%>
+
+<div class="<%- self.generateClassName('form', this.modifiers) %>">
+    <div class="form__body">
+        <div class="form__bodypart form__textpart">
+            <h4><%= i18n('Черновик устарел') %></h4>
```
Comment (zouxuoz) [inline src/blocks/form/TransformOldDraft/index.ejs]: should be escaped

File: src/blocks/form/TransformOldDraft/index.ejs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                        label: i18n('Создать маркетинговый черновик'),
+                        value: 'marketing',
+                        checked: true
+                    })%>
+                </div>
+                <div>
+                    <%= this.include(RadioButton, {
+                        name: 'type',
+                        value: 'transactional',
+                        label: i18n('Создать письмо для автоматизации'),
```
Comment (zouxuoz) [inline src/blocks/form/TransformOldDraft/index.ejs]: unnecessary comma

File: src/blocks/form/TransformOldDraft/index.ejs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+                    })%>
+                </div>
+                <div>
+                    <%= this.include(RadioButton, {
+                        name: 'type',
+                        value: 'template',
+                        label: i18n('Создать шаблон'),
```
Comment (zouxuoz) [inline src/blocks/form/TransformOldDraft/index.ejs]: unnecessary comma

File: src/blocks/table/mailings/mailings.js [hunk 7]
```diff
@@ -21,7 +22,7 @@ define(function(require) {
                 var status = model.get('status'),
                     field = '';
 
-                if(model instanceof DraftModel)
+                if(model instanceof DraftModel || model instanceof OldDraft)
```
Comment (zouxuoz) [inline src/blocks/table/mailings/mailings.js]: missed brackets for `if` statement

File: src/blocks/table/mailings/row.ejs [hunk 8]
```diff
@@ -14,7 +15,7 @@
         })%>
     <% } %>
     
-<% } else if(draft instanceof draftModel) { %>
+<% } else if(draft instanceof draftModel || draft instanceof OldDraft) { %>
```
Comment (zouxuoz) [inline src/blocks/table/mailings/row.ejs]: missed whitespace after `if`

File: src/pages/mailings/mailing/transform/title.ejs [hunk 9]
```diff
@@ -0,0 +1,15 @@
+<div class="header__section header__title">
+	<div class="header__content">
+        <div class="header__backtitle"><a class="navigation__link" href="/mailings/manage"><%= UI.icons.back%><%=i18n('Назад к списку рассылок')%></a></div>
```
Comment (zouxuoz) [inline src/pages/mailings/mailing/transform/title.ejs]: text should be escaped

File: src/blocks/form/TransformOldDraft/ActionPanel.less [hunk 10]
```diff
@@ -0,0 +1,16 @@
+@import 'src/kit/pallete';
+
+.ActionPanel {
```
Comment (rosendi) [inline src/blocks/form/TransformOldDraft/ActionPanel.less]: Why is it uppercased? `actionPanel` is the correct name.

File: src/blocks/form/TransformOldDraft/ExpandableInfo.css [hunk 11]
```diff
@@ -0,0 +1,17 @@
+.ExpandableInfo {
```
Comment (rosendi) [inline src/blocks/form/TransformOldDraft/ExpandableInfo.css]: Same here `expandableInfo`

File: src/blocks/form/TransformOldDraft/ActionPanel.less (hunk 10 above)
Comment (rosendi) [inline src/blocks/form/TransformOldDraft/ActionPanel.less]: It's BEM.

File: src/blocks/form/TransformOldDraft/ExpandableInfo.ejs [hunk 12]
```diff
@@ -0,0 +1,17 @@
+
+<div class="<%- this.generateClassName('ExpandableInfo', this.modifiers) %>">
+    <div class="ExpandableInfo__toggle">
+        <span class="js-toggle">
+            <%= this.expanded ? '▼' : '►'%> <%= i18n('Что это означает?') %>
```
Comment (rosendi) [inline src/blocks/form/TransformOldDraft/ExpandableInfo.ejs]: haha

File: src/blocks/form/TransformOldDraft/ExpandableInfo.js [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+        defaults: {
+            expanded: false
+        },
+
+        events: {
+            'click .js-toggle': '_toggle'
+        },
+
+        _toggle: function() {
+            this.expanded = !this.expanded;
```
Comment (rosendi) [inline src/blocks/form/TransformOldDraft/ExpandableInfo.js]: Are we using the `expanded` flag in the other places?

File: src/blocks/form/TransformOldDraft/ExpandableInfo.less [hunk 14]
```diff
@@ -0,0 +1,23 @@
+@import 'src/kit/pallete';
+
+.ExpandableInfo {
```
Comment (rosendi) [inline src/blocks/form/TransformOldDraft/ExpandableInfo.less]: Same here `expandableInfo`

File: src/blocks/form/TransformOldDraft/ScaledDraftPreview.less [hunk 15]
```diff
@@ -0,0 +1,11 @@
+.ScaledDraftPreview {
```
Comment (rosendi) [inline src/blocks/form/TransformOldDraft/ScaledDraftPreview.less]: Same here `scaledDraftPreview`

File: src/blocks/form/TransformOldDraft/index.js [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+        events: {
+            'click .js-transform': 'submit',
+            'change [name="type"]': '_handleTypeChange'
+        },
+
+        _handleTypeChange: function(event) {
+            var message = this.el.querySelector('.js-message').block;
+            var value = event.target.value;
+
+            message.props.type = value;
```
Comment (rosendi) [inline src/blocks/form/TransformOldDraft/index.js]: Just message.props.type =  event.target.value;

File: src/blocks/form/TransformOldDraft/index.js [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+            message.render();
+        },
+
+        save: function(data) {
+            var draft = this.props.draft;
```
Comment (rosendi) [inline src/blocks/form/TransformOldDraft/index.js]: Add a new line

File: src/blocks/form/TransformOldDraft/index.js [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+            switch(data.type) {
+                case 'marketing':
+                    return draft.createMarketing().then(function(newModel) {
+                        app.navigate('/mailings/mailing/create?id=' + newModel.get('id'));
+                    });;
```
Comment (rosendi) [inline src/blocks/form/TransformOldDraft/index.js]: Add new line before each `case`

File: src/blocks/form/TransformOldDraft/index.js [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+                    });; 
+                case 'transactional':
+                    return draft.createTransactional().then(function(newModel) {
+                        app.navigate('/automation/letters/wizard?id=' + newModel.get('id'));
+                    });;
+                case 'template':
+                    return draft.createTemplate().then(function(newModel) {
+                        app.navigate('/mailings/templates/edit?id=' + newModel.get('id'));
+                    });
+            }
```
Comment (rosendi) [inline src/blocks/form/TransformOldDraft/index.js]: New line

File: src/blocks/mailing/draftMailing/draftMailing.ejs [hunk 20]
```diff
@@ -3,18 +3,23 @@
...
+            <% } else { %>
+                <%= this.draft.getType() === 'block' ? blockIcon : htmlIcon %>
+            <% } %>
         </span>
 
     </td>
     <td>
         <div class="mailing__primaryData">
-            <% var name = self.draft.get('name') %>
+            <% var name = self.draft.get('name'); %>
```
Comment (rosendi) [inline src/blocks/mailing/draftMailing/draftMailing.ejs]: Why? Do you need creating a new var? Why not just use `self.draft.get('name');`

File: src/blocks/mailing/draftMailing/draftMailing.ejs [hunk 21]
```diff
@@ -24,13 +29,16 @@
...
-            text: i18n('Редактировать'),
-            href: self._constructUrl(),
-            modifiers: {
-                color: 'gray'
-            }
-        }) %>
+        <% if (!this._isOld()) { %>
+            <%= self.include(UI.button, {
+                text: i18n('Редактировать'),
+                href: self._constructUrl(),
```
Comment (rosendi) [inline src/blocks/mailing/draftMailing/draftMailing.ejs]: new line

File: src/blocks/mailing/draftMailing/draftMailing.js [hunk 22]
```diff
@@ -21,8 +24,18 @@ define(function(require) {
         },
 
         _constructUrl: function() {
-            var mode = this.draft.isFilled() ? 'edit' : 'create';
+            var mode;
```
Comment (rosendi) [inline src/blocks/mailing/draftMailing/draftMailing.js]: new line

File: src/blocks/table/mailings/row.ejs [hunk 23]
```diff
@@ -3,18 +3,19 @@
     var draft = data.draft;
     var draftModel = require('resources/marketingDraft/model'),
         issueModel = require('resources/_issue/model');
```
Comment (rosendi) [inline src/blocks/table/mailings/row.ejs]: var

File: src/blocks/table/mailings/row.ejs [hunk 24]
```diff
@@ -3,18 +3,19 @@
...
+    var OldDraft = require('resources/oldDraft/model');
 %>
 
-<% if(draft instanceof issueModel) { %>
+<% if (draft instanceof issueModel) { %>
 
-    <%if(draft.get('state') !== 'unknown') { %>
+    <%if (draft.get('state') !== 'unknown') { %>
         <%= this.include(require('blocks/mailing/issueMailing/issueMailing'), {
```
Comment (rosendi) [inline src/blocks/table/mailings/row.ejs]: The require to the top.

File: src/kit/pallete.less [hunk 25]
```diff
@@ -32,7 +32,8 @@
 @lightblue: #5bb5cb;
 @deepblue: #1E88E5;
 @deepgreen: #4caf50;
-
+@weakblue: #E2F2FE;
```
Comment (rosendi) [inline src/kit/pallete.less]: FUCK ME

File: src/pages/mailings/mailing/transform/title.ejs [hunk 26]
```diff
@@ -0,0 +1,15 @@
+<div class="header__section header__title">
```
Comment (rosendi) [inline src/pages/mailings/mailing/transform/title.ejs]: Don't use tab, use the spaces. Check that ejs.

File: src/resources/oldDraft/collection.js [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+
+    return Collection.extend({
+        model: require('./model'),
+
+        action: 'issue.draft.list',
+
+        fetch: function(options) {
+            var self = this;
+
+            if (this.fetching)
```
Comment (rosendi) [inline src/resources/oldDraft/collection.js]: Put the brackets.

File: src/resources/oldDraft/collection.js [hunk 28]
```diff
@@ -0,0 +1,39 @@
...
+                return this.deferred.promise();
+
+            var deferred = api.call({
```
Comment (rosendi) [inline src/resources/oldDraft/collection.js]: Check the style here.

File: src/resources/oldDraft/model.js [hunk 29]
```diff
@@ -0,0 +1,105 @@
+define(function(require) {
+
+    var Model = require('kit/model/model');
+    var	DraftBlockPresentation = require('resources/presentation/draft/block/model');
```
Comment (rosendi) [inline src/resources/oldDraft/model.js]: Does it work?????

File: src/resources/oldDraft/model.js [hunk 30]
```diff
@@ -0,0 +1,105 @@
...
+    var	DraftHTMLPresentation = require('resources/presentation/draft/html/model');
```
Comment (rosendi) [inline src/resources/oldDraft/model.js]: And here....

**Comment by rosendi** [comment]
LGTM

**Pull request merged**